dev-util/Tensile/Tensile-5.0.2.ebuild

```diff
@@ -0,0 +1,87 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{8..10} )
+DISTUTILS_USE_PEP517=setuptools
+inherit distutils-r1
+
+DESCRIPTION="Stretching GPU performance for GEMMs and tensor contractions"
+HOMEPAGE="https://github.com/ROCmSoftwarePlatform/Tensile"
+SRC_URI="https://github.com/ROCmSoftwarePlatform/Tensile/archive/rocm-${PV}.tar.gz -> rocm-Tensile-${PV}.tar.gz
+		https://github.com/littlewu2508/littlewu2508.github.io/raw/main/gentoo-distfiles/${P}-PR1419.patch.gz"
+S="${WORKDIR}/${PN}-rocm-${PV}"
+
+LICENSE="MIT"
+KEYWORDS="~amd64"
+SLOT="0/$(ver_cut 1-2)"
+
+# Not compatible with recent versions of pytest
+RESTRICT="test"
```

Comment:
Have you tested if the tests work in this release? It's a lot newer so hopefully it works with the in-tree versions of pytest.

Reply:
OK let me have a try. Currently I can enable tests but there are a lot of failures. The main difficulties are:

1. Tensile require some Source files and we installed it into /usr/share/Tensile which is not in the site-packages directory, so I hard-coded the search path in Tensile.py, __init__.py, etc. But when running test, Tensile is not merged and /usr/share/Tensile doesn't exists.
2. Tensile's sub-modules like `Components` are handled via `python_domodule` in `python_install`, because I removed 'package_data' in `setup.py` to avoid installing hundreds of configurations file in site-package directory. However, when running `src_test`, `src_install` is not invoked yet, and these sub-modules are missing.

I'm trying to figure out solutions for the two problems.

Reply:
Some work is done by enabling tests, located at https://github.com/littlewu2508/gentoo/tree/Tensile-5-try-test. However I don't put testing Tensile as a high-priority target, since its mainly a benchmark and build tool for rocBLAS, and in my use cases nothing breaks.